src/xclient_zmq.cpp

```diff
@@ -0,0 +1,50 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xeus-zmq/xclient_zmq.hpp"
+#include "xclient_zmq_impl.hpp"
+
+namespace xeus
+{
+
+    xclient_zmq::xclient_zmq(xcontext& context,
+                            const std::string& user_name,
+                            const xeus::xconfiguration& config)
+        : p_client_impl(std::make_unique<xclient_zmq_impl>(context.get_wrapped_context<zmq::context_t>(), user_name, config))
```

Comment:
I think the lower level classes like the channel clients should deal with `zmq::context`, so we could either go from `xcontext` to `zmq::context` here or we could maintain `xcontext` in `xclient_zmq_impl` and take care of it in the level below.

Reply:
Going to `zmq::context` here is fine, these are implementation headers and they're not exposed to "consumers" of `xeus-zmq`

Reply:
Noted.